### PR TITLE
Feature add locale

### PR DESCRIPTION
**Pull request by boldtrn**
This PR adds the locale to the Response. We parse the given locale if it's valid. If it's valid we forward it to the geocoding provider, otherwise we fallback to English.

@karussell Would you mind to review this PR?

### CODE REVIEW

File: src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java [hunk 1]
```diff
@@ -32,4 +34,31 @@ void checkInvalidParameter(boolean reverse, String query, String point) {
...
+        if (isValid(lo)) {
+            return lo.toLanguageTag();
+        } else {
+            return "en";
+        }
+    }
+
+    // Taken from: http://stackoverflow.com/a/3684832/1548788
+    Locale parseLocale(String locale) {
+        String[] parts = locale.split("_");
```
Comment (karussell) [inline src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java:48]: Here we could use additionally `locale.replace("-", "_").split("_");`

**Review by karussell — APPROVED** [review]
LGTM

File: src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java (hunk 1 above)
Comment (boldtrn) [reply src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java:48]: Thanks for the comment. I think I found an even better way to do this. There is a function in the Locale class to do this: https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#forLanguageTag(java.lang.String)

I added some more tests for it

**Comment by boldtrn** [comment]
Thanks for your review :)

**Pull request merged**